### PR TITLE
spec: device: add initial device protocol specification

### PR DESCRIPTION
**Pull request by FFY00**
Signed-off-by: Filipe Laíns <lains@archlinux.org>

### CODE REVIEW

**Comment by FFY00** [comment]
Preview: https://github.com/FFY00/openinput-protocol/tree/dev-spec

Github's rst renderer is not great, it does not play well for eg. with the `note` directive. That is fine, in time I want to generate a documentation website with sphinx (probably using readthedocs).

File: specification/device-protocol/README.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    | description | 0x10 | Function Page | Function ID |      Data     |
+    +-------------+------+---------------+-------------+---------------+
+
+
+.. table:: Table 5 - Long packet structure
+
+    +-------------+------+---------------+-------------+---+---+---+---+---+---+---+----+----+----+----+----+----+----+----+----+
+    |     byte    |   0  |       1       |      2      | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 |
+    +=============+======+===============+=============+========================================================================+
+    | description | 0x10 | Function Page | Function ID |                                 Data                                   |
```
Comment (perigoso) [inline specification/device-protocol/README.rst]: 0x21

File: specification/device-protocol/README.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    requests they hold function arguments, and for responses, they hold return
+    values.
+
+
+.. table:: Table 4 - Short packet structure
+
+    +-------------+------+---------------+-------------+---+---+---+---+
+    |     byte    |   0  |       1       |      2      | 3 | 4 | 5 | 6 |
+    +=============+======+===============+=============+===============+
+    | description | 0x10 | Function Page | Function ID |      Data     |
```
Comment (perigoso) [inline specification/device-protocol/README.rst]: 0x20

**Review by perigoso — CHANGES_REQUESTED** [review]
Looks good :+1: just a couple tiny typos

**Comment by FFY00** [comment]
Ups.

**Review by perigoso — APPROVED** [review]

**Pull request merged**